framework/assets/yii.gridView.js

```diff
@@ -100,10 +108,12 @@
 
             $grid.find('form.gridview-filter-form').remove();
             var $form = $('<form action="' + url + '" method="get" class="gridview-filter-form" style="display:none" data-pjax></form>').appendTo($grid);
-            $.each(data, function (name, value) {
-                $form.append($('<input type="hidden" name="t" value="" />').attr('name', name).val(value));
+            $.each(data, function (name, values) {
+                $.each(values, function (index, value) {
+                    $form.append($('<input type="hidden" name="t" value="" />').attr('name', name).val(value));
```

Comment:
That's true. But this change is only required when you use some kind of input, that allows you to pass multiple values into it. For example you can use select with `multiple` set to true.
Of couse setting correct name for input is developer's task, but if you use for example `Html::listBox` method it will add `[]` to the name automatically, same think will happen if you use some widget based on the list box (for example one of select2 widgets). If you use checkboxes in the filter, then `[]` will be added too.